app/public/cantusdata/helpers/expandr.py

```diff
@@ -1,152 +1,148 @@
     mode_output = []
-    if "1" in input_list:
-        mode_output.append("1")
-    if "2" in input_list:
-        mode_output.append("2")
-    if "3" in input_list:
-        mode_output.append("3")
-    if "4" in input_list:
-        mode_output.append("4")
-    if "5" in input_list:
-        mode_output.append("5")
-    if "6" in input_list:
-        mode_output.append("6")
-    if "7" in input_list:
-        mode_output.append("7")
-    if "8" in input_list:
-        mode_output.append("8")
-    if "*" in input_list:
-        mode_output.append("No music")
-    if "r" in input_list:
-        mode_output.append("Formulaic")
-    if "?" in input_list:
-        mode_output.append("Uncertain")
-    if "S" in input_list:
-        mode_output.append("Responsory (special)")
-    if "T" in input_list:
-        mode_output.append("Chant in Transposition")
+    for char in mode_code_stripped:
+        if char in "12345678":
+            mode_output.append(char)
+            continue
+        match char:
+            case "*":
+                mode_output.append("No music")
+            case "r":
+                mode_output.append("Formulaic")
+            case "?":
+                mode_output.append("Uncertain")
+            case "S":
+                mode_output.append("Responsory (special)")
+            case "T":
+                mode_output.append("Chant in Transposition")
     outstring = " ".join(mode_output)
     return outstring
 
 
-def expand_genre(genre_code):
-    if genre_code in genres:
-        description = genres[genre_code]
-        # some extra stuff in parentheses is showing up
-        paren = description.find("(")
-        return description[: paren - 1] if paren != -1 else description
+class GenreExpander:
+    """
+    Loads the genre mapping from the CantusDB API and provides a method to retrieve
+    the full text genre description based on the given genre code.
+    """
 
-    # If nothing was found, return the original
-    return genre_code
+    cantus_db_api_endpoint = "https://cantusdatabase.org/genres"
+    request_headers = {"Accept": "application/json"}
 
+    def __init__(self) -> None:
+        self.genre_data = self.load_genre_data()
 
-def expand_differentia(differentia_code):
+    def load_genre_data(self) -> dict[str, str]:
+        """
+        Loads the genre list from the CantusDB API and returns a dictionary mapping
+        genre codes to genre descriptions.
+        """
+        response = requests.get(
+            self.cantus_db_api_endpoint, headers=self.request_headers, timeout=5
+        )
+        response.raise_for_status()
+        genre_map: dict[str, str] = {
+            x["name"]: x["description"] for x in response.json()["genres"]
+        }
+        return genre_map
+
+    def expand_genre(self, genre_code: str) -> str:
+        """
+        Gets the genre description based on the genre code.
+        """
+        if genre_code in self.genre_data:
```

Comment:
It's generally better to exit early. It makes the default path clearer, and removes an indent.

```py
if genre_code not in self.genre_data:
    return genre_code

description = self.genre_data[genre_code]
# some extra stuff in parentheses is showing up
paren = description.find("(")
return description[: paren - 1] if paren != -1 else description
```